grails-app/domain/org/pih/warehouse/product/ProductPackage.groovy

```diff
@@ -56,7 +56,15 @@ class ProductPackage implements Comparable<ProductPackage>, Serializable {
         description(nullable: true)
         gtin(nullable: true)
         uom(nullable: true)
-        product(nullable: true, unique: ['uom', 'quantity', 'productSupplier'])
+        product(nullable: true, validator: { Product product, ProductPackage obj ->
+            ProductPackage productPackage = findWhere(
+                    uom: obj.uom,
+                    product: product,
+                    quantity: obj.quantity,
+                    productSupplier: obj.productSupplier
+                    )
+            return !productPackage || productPackage?.id == obj?.id
+        })
```

Comment:
I had to refactor this part, because otherwise packages:
1. productSupplier: null, uom: "XYZ", quantity: X
2. productSupplier: "ZYX", uom: "XYZ", quantity: X

did not pass the unique constraint, because when no `productSupplier` provided, for the validation query it was not searching for eventual productPackage with `productSupplier` as `null`, but skipped it while quering, so for the uniqness check, it was doing:
```
select from product_package where uom = ... and product = ... and quantity = ...
```
instead of:
```
select from product_package where uom = ... and product = ... and quantity = ... and product_supplier IS NULL
```